Api match mds schema

tacaswell opened the pull request.

tacaswell committed:
MNT: use doct instead of doc

This is where it started, but time to use the external library.

tacaswell committed:
MNT: fully remove mongoengine + match MDS schema

 - use our own module-level singleton to manage DB state
 - use uid, not oid, for foreign key reference

tacaswell committed:
BLD: update conda-recipe

ericdill commented on .travis.yml:
```diff
@@ -24,7 +24,7 @@ install:
 - conda create -n testenv pip nose python=$TRAVIS_PYTHON_VERSION pymongo six pyyaml numpy pandas jinja2 boltons prettytable humanize
 - source activate testenv
-- pip install mongoengine
+- 'pip install https://github.com/NSLS-II/doct/zipball/master#egg=doct'
```
You can `conda install doct` now. I built master and uploaded it to `lightsource2`

tacaswell committed:
TST: use conda for doct dep

tacaswell committed:
FIX: fix major typo

tacaswell committed:
TST: add asserts that data round-trips correctly

tacaswell committed:
SCHEMA: separate data and timestamps in database

 - Saves transforms going in and out.
 - Most insertions should happen via BlueSky or similar which can ensure
   that the data and timestamp keys stay in sync

tacaswell committed:
STY: whitespace

tacaswell committed:
API: restore exporting doct as doc at toplevel

To maintain back-compatibility.

tacaswell committed:
TST: add custom warning test

tacaswell committed:
FIX: fix typos in connection management

tacaswell committed:
MNT: finish removing custom
